applications/Chat/coati/trainer/strategies/base.py

```diff
@@ -111,7 +111,7 @@ def unwrap_model(model: nn.Module) -> nn.Module:
         return model
 
     def save_model(self, model: nn.Module, path: str, only_rank0: bool = True, **kwargs) -> None:
-        self.booster.save_model(model, path, shard=not only_rank0, **kwargs)
+        self.booster.save_model(model, path, shard=False, **kwargs)
```

Comment:
Why not shard?

Reply:
We need to replace the keys in a checkpoint by removing the "model" prefix and saving it as unsharded. Only one file needs to be replaced. To save it in sharded format can be included in the next pull request.